### PR TITLE
add orders timed update jobs

### PR DESCRIPTION
**Pull request by bigdaddyrus**

### CODE REVIEW

File: src/controller/helper/orderUpdater.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { getRepository } from "typeorm";
+import { ListingStatus } from "../../entities/Items";
+
+const ORDER_AUTO_COMPLETE_DAYS = 5;
+const ORDER_AUTO_CANCEL_MINUTES = 1;
+const UPDATE_MAX = 50; // set limite to avoid hitting max connections
+
+export const autoCompleteOrders = async (): Promise<void> => {
+  const orders = await Orders.find({status: OrderStatus.SHIPPED});
+
```
Comment (OrangeChickens) [inline src/controller/helper/orderUpdater.ts:13]: return if orders.length  === 0?

File: src/controller/helper/orderUpdater.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    return; 
+  }
+  
+  const results = orders.map(async order => {
+    const orderCreateDate = new Date(order.createdtime);
+    const orderCancelTime = moment(orderCreateDate).add(ORDER_AUTO_CANCEL_MINUTES, "m").toDate();
+    if (orderCancelTime < new Date()) {
+      order.status = OrderStatus.CANCELLED;
+      // Relist corresponding items
+      const results = order.orderItems.map(async item => {
```
Comment (OrangeChickens) [inline src/controller/helper/orderUpdater.ts:46]: rename to orderItemResults? there are two results named here.

File: src/controller/helper/orderUpdater.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    .leftJoinAndSelect("orders.orderItems", "items")
+    .where({status: OrderStatus.OPEN})
+    .take(UPDATE_MAX)
+    .getMany();
+
+  if (orders.length === 0) {
+    return; 
+  }
+  
+  const results = orders.map(async order => {
```
Comment (OrangeChickens) [inline src/controller/helper/orderUpdater.ts:40]: for cancelled orders we can query directly ? .where(time < xx)

File: src/controller/helper/orderUpdater.ts (hunk 1 above)
Comment (bigdaddyrus) [reply src/controller/helper/orderUpdater.ts:13]: 其实不用 下面那个加下是减少log数量

File: src/controller/helper/orderUpdater.ts (hunk 3 above)
Comment (bigdaddyrus) [reply src/controller/helper/orderUpdater.ts:40]: 有但是也比较麻烦
https://stackoverflow.com/questions/53495716/searching-data-older-than-a-date-with-typeorm

File: src/controller/helper/orderUpdater.ts (hunk 2 above)
Comment (bigdaddyrus) [reply src/controller/helper/orderUpdater.ts:46]: 可

**Pull request merged**